src/controllers/untitledSqlDocumentService.ts

```diff
@@ -18,25 +18,16 @@ export default class UntitledSqlDocumentService {
      * Creates new untitled document for SQL query and opens in new editor tab
      * with optional content
      */
-    public newQuery(content?: string): Promise<Uri> {
-        return new Promise<Uri>(async (resolve, reject) => {
-            try {
-
-                // Open an untitled document. So the  file doesn't have to exist in disk
-                let doc = await this.vscodeWrapper.openMsSqlTextDocument(content);
-
-                // Show the new untitled document in the editor's first tab and change the focus to it.
-                await this.vscodeWrapper.showTextDocument(doc, 1, false);
-                const editor = vscode.window.activeTextEditor;
-                const position = editor.selection.active;
-                let newPosition = position.with(position.line + 1, 0);
-                let newSelection = new vscode.Selection(newPosition, newPosition);
-                editor.selection = newSelection;
-                resolve(doc.uri);
-            } catch (error) {
-                reject(error);
-            }
-        });
+    public async newQuery(content?: string): Promise<vscode.TextEditor> {
+        // Open an untitled document. So the  file doesn't have to exist in disk
+        let doc = await this.vscodeWrapper.openMsSqlTextDocument(content);
+        // Show the new untitled document in the editor's first tab and change the focus to it.
+        const editor = await this.vscodeWrapper.showTextDocument(doc, 1, false);
+        const position = editor.selection.active;
+        let newPosition = position.with(position.line + 1, 0);
+        let newSelection = new vscode.Selection(newPosition, newPosition);
```

Comment:
I see lots of +1's around in this code but no explanation of why. Would be good to try and add comments since that's not something I'd consider normal and it's not clear why they're needed